cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -347,10 +349,51 @@ def charm_status(kube_control):
     update_kubelet_status()
 
 
+def deprecated_extra_args():
+    '''Returns a list of tuples (config_key, arg) for args that have been set
+    via extra-args, but are deprecated.
+
+    This works by parsing help output, which can be brittle. Be cautious when
+    calling this.
+    '''
+    deprecated_args = []
+    services = [
+        # service       config_key
+        ('kubelet',    'kubelet-extra-args'),
+        ('kube-proxy', 'proxy-extra-args')
+    ]
+    for service, config_key in services:
+        # Parse help output into a format we can check easily
+        cmd = [service, '-h']
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        sections = re.split('\n\s*--', output.decode('utf-8'))[1:]
+        partitioned_sections = [section.partition(' ') for section in sections]
+        arg_help = {part[0]: part[2] for part in partitioned_sections}
+
+        # Check extra-args against the help output
+        extra_args = parse_extra_args(config_key)
+        for arg in extra_args:
+            if arg in arg_help and 'DEPRECATED:' in arg_help[arg]:
+                deprecated_args.append((config_key, arg))
```

Comment:
Should we call out invalid args here? Seems if arg is not in arg_help there is a problem.

Reply:
Not necessarily. Kubelet 1.8.x had a hidden `experimental-dockershim` flag that didn't show in help output, for example.

I don't see any hidden args in 1.9.x, 1.10.x or 1.11.x, but I don't want to assume that will always be the case.

Might be worthwhile to log a warning though. What do you think?